src/app/GitUI/CommandsDialogs/Menus/WorkingDirectoryToolStripSplitButton.cs

```diff
@@ -0,0 +1,181 @@
+using GitCommands;
+using GitCommands.UserRepositoryHistory;
+using GitExtensions.Extensibility.Git;
+using GitExtensions.Extensibility.Translations;
+using GitUI.CommandsDialogs.BrowseDialog;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs.Menus
+{
+    /// <summary>
+    ///  Represents a split button that contains the recent repositories.
+    /// </summary>
+    internal class WorkingDirectoryToolStripSplitButton : ToolStripSplitButton, ITranslate
+    {
+        private readonly TranslationString _noWorkingFolderText = new("No working directory");
+        private readonly TranslationString _configureWorkingDirMenu = new("&Configure this menu...");
+
+        private Func<IGitUICommands>? _getUICommands;
+        private IRepositoryHistoryUIService _repositoryHistoryUIService;
+
+        // NOTE: This is pretty bad, but we want to share the same look and feel of the menu items defined in the Start menu.
+        private StartToolStripMenuItem _startToolStripMenuItem;
+        private ToolStripMenuItem _closeToolStripMenuItem;
+
+        public WorkingDirectoryToolStripSplitButton()
+        {
+            Name = nameof(WorkingDirectoryToolStripSplitButton);
+
+            Image = Properties.Resources.RepoOpen;
+            ImageAlign = ContentAlignment.MiddleLeft;
+            ImageTransparentColor = Color.Magenta;
+            TextAlign = ContentAlignment.MiddleLeft;
+        }
+
+        /// <summary>
+        ///  Gets the current instance of the git module.
+        /// </summary>
+        protected IGitModule Module => UICommands.Module;
+
+        /// <summary>
+        ///  Gets the form that is displaying the menu item.
+        /// </summary>
+        protected static Form? OwnerForm
+            => Form.ActiveForm ?? (Application.OpenForms.Count > 0 ? Application.OpenForms[0] : null);
```

Comment:
We should not need to implement this for every user control anew.

Reply:
Yes and no. Each user control that needs to show a dialog needs to correctly determine "owner". But given that all functionality relies on statics, we could conceivably abstract this behind an injectable service (to make it testable). E.g., something like:
```cs
public interface IDialogOwnerFormPovider
{
    Form? FindOwner();
}

internal sealed class DialogOwnerFormPovider : IDialogOwnerFormPovider
{
    public Form? FindOwner() => Form.ActiveForm ?? (Application.OpenForms.Count > 0 ? Application.OpenForms[0] : null);
}
```

Reply:
As it is `static`, this function can be provided by a static helper class. I do not see a need for an interface.

Reply:
Then it becomes very test unfriendly